packages/console/src/components/AppContent/components/UserInfo/index.tsx

```diff
@@ -0,0 +1,80 @@
+import { useLogto } from '@logto/react';
+import { User, UserRole } from '@logto/schemas';
+import classNames from 'classnames';
+import React, { useEffect, useRef, useState } from 'react';
+import { useTranslation } from 'react-i18next';
+
+import Dropdown, { DropdownItem } from '@/components/Dropdown';
+import { getAvatarById } from '@/consts/avatars';
+import useApi from '@/hooks/use-api';
+import SignOut from '@/icons/SignOut';
+
+import * as styles from './index.module.scss';
+
+const UserInfo = () => {
+  const { isAuthenticated, signOut, getIdTokenClaims } = useLogto();
+  const api = useApi();
+  const { t } = useTranslation(undefined, { keyPrefix: 'admin_console' });
+  const anchorRef = useRef<HTMLDivElement>(null);
+  const [showDropDown, setShowDropdown] = useState(false);
+  const [user, setUser] = useState<User>();
+
+  useEffect(() => {
+    (async () => {
+      if (isAuthenticated) {
+        const { sub: userId } = getIdTokenClaims() ?? {};
+
+        if (userId) {
+          const data = await api.get(`/api/users/${userId}`).json<User>();
+          setUser(data);
+        }
+      }
+    })();
+  }, [api, isAuthenticated, getIdTokenClaims]);
+
+  if (!user) {
+    return null;
+  }
+
+  const { id, name, avatar, roleNames } = user;
+  const isAdmin = roleNames.includes(UserRole.Admin);
+
+  return (
+    <>
+      <div
+        ref={anchorRef}
+        className={classNames(styles.container, showDropDown && styles.active)}
+        onClick={() => {
+          setShowDropdown(true);
+        }}
+      >
+        <img src={avatar ?? getAvatarById(id)} />
+        <div className={styles.wrapper}>
+          <div className={styles.name}>{name}</div>
+          {isAdmin && <div className={styles.role}>{t('user_details.roles.admin')}</div>}
+        </div>
+      </div>
+      <Dropdown
+        anchorRef={anchorRef}
+        className={styles.dropdown}
+        isOpen={showDropDown}
+        horizontalAlign="end"
+        onClose={() => {
+          setShowDropdown(false);
+        }}
+      >
+        <DropdownItem
+          className={styles.dropdownItem}
+          icon={<SignOut />}
+          onClick={() => {
+            void signOut(`${window.location.origin}/console/sign-out`);
```

Comment:
Oh, yes. This is actually some of my uncommitted local code. I was actually having a similar idea of having a separate route in AC (which didn't work). But you are talking about the server side anonymous route... and it might work this time. Will try

Reply:
Well, problem solved. We can't call `signOut()` from here, as the `AppContent` is still mounting and it will automatically trigger `signIn` right after the `isAuthenticated` turned false.

So instead, we navigate user to a `/sign-out` route, which is located at the same level as `AppContent`. This will unmount `AppContent`, and thus calling `signOut` will not be interrupted any more.

Please refer to the gif I attached. Now sign out is working as expected.